### ISSUE TITLE
[Auditbeat] MacOS auditing

### ISSUE THREAD

**Issue by andrewkroh** [elastic/beats#6061]
There is some built-in auditing support in macOS. I don't know much about it yet, but it sounds like we can get execve info at a minimum. This needs more investigation.
- https://developer.apple.com/legacy/library/documentation/Darwin/Reference/ManPages/man2/auditon.2.html
- https://ilostmynotes.blogspot.com/2013/10/openbsm-auditd-on-os-x-these-are-logs.html
- https://github.com/objective-see/ProcInfo/blob/e6c01fdf69a605491aa52d7466888ef1f3f2e984/procInfo/ProcessMonitor.m#L125
 - `sudo praudit -l /dev/auditpipe`

**Comment by asekhar** [comment]
Does the new version include support for auditd on OSX? Wondering what the status of this issue is.

**Comment by cwurm** [comment]
@asekhar Not yet. Though we are working on some enhancements to Auditbeat that will allow to collect additional information on macOS as well.

If you don't mind me asking - what information would you want to collect?

Note: macOS does not have `auditd` as Linux does and so we will always be limited in what we can collect compared to it (in Linux, you can pretty much collect anything).

**Comment by asekhar** [comment]
I'd like to get a close to the artifacts described here: https://blogs.dropbox.com/tech/2018/04/4696/ as possible, but among others, process name, path, arguments, parent process, network connections, file creations.

**Comment by Vladimir-csp** [comment]
+1 for native FreeBSD audit support, with perspective of abstracting the data together with Linux audit events

**Comment by gwsales** [comment]
This will get you xml, not sure if it would be easier to pass into a parser or if it adds any additional load.
`sudo praudit -x -l /dev/auditpipe`

**Comment by deepybee** [comment]
@andrewkroh Auditbeat is now listed as supported in our [Support Matrix](https://www.elastic.co/support/matrix) for MacOS >= 10.13 (High Sierra). Can this issue be closed?

**Comment by andrewkroh** [comment]
No, it cannot be closed. There are many features of Auditbeat that work on MacOS, but reading from MacOS audit data from the auditpipe is not one of them yet.

**Comment by deepybee** [comment]
OK, thanks for the clarification!

**Comment by gwsales** [comment]
@andrewkroh can you get this one reopened or maybe assigned to a team?

**Comment by inqueue** [comment]
Passing on these links on behalf of another:

- https://community.spiceworks.com/topic/562291-how-to-audit-log-file-access-events-on-mac-os-x
- https://www.scip.ch/en/?labs.20150108

The first should be covered by the File Integrity Module. The second appears to be this GH issue.

**Comment by jasonborchardt** [comment]
Can this issue be re-opened? It appears auditbeat attempts to parse process information in real time instead of subscribing to events in MacOS, which causes many events to be missed if they start and stop quickly.  Additionally, in order to get information about processes executing from auditd, you must modify files in /etc/security, then reboot the system (as SIP prevents you from restarting the auditd process with the new configuration).  Additionally, every OS update wipes out changes to files located in /etc/security.

It is possible to build filters and subscribe to system events, especially in MacOS 11.x.  Other tools such as Famf protect do this.

**Comment by a03nikki** [comment]
@elastic/siem : Could we please get this ticket re-opened?  It has not been resolved yet and is important functionality.

**Comment by austinsonger** [comment]
Agreed..

**Comment by MakoWish** [comment]
No movement on this? We are only just now starting to test Auditbeat on Mac (we are mainly a Windows and Linux shop).

**Comment by MakoWish** [comment]
For us it is mainly authentication events.

**Comment by asekhar** [comment]
process executions, network connections, file creation and authentication events

**Comment by gwsales** [comment]
* Authentication
* Process Execution / Process Creation
* Network Connections
* File Reads and Writes
* System Changes

Would be nice to add some advanced things as well like monitoring plist / ssh keys / system startup / listening processes. It's been awhile since I've looked at Auditbeat on Mac, we've mostly moved onto other tools since this doesn't seem to be getting much traction or progress.

**Comment by gwsales** [comment]
@elastic/siem Has there been any progress here? Elastic has solid coverage on Windows and Linux, but really seemed to just skip over anything MacOS. 

Would be great to see some updates here.

**Comment by MakoWish** [comment]
Yeah, almost six years later, and no traction.

**Comment by leandrojmp** [comment]
Hello,

Now that the UnifiedLogs integration was created, can this issue get some traction? We are trying to create some filters to get Authentication and User Changes logs since without any filters this integration generates an insane amount of logs.

It would be nice to have some default filters configured for logs from Authentication, User Changes, System Changes etc.

**Comment by pierrehilbert** [comment]
@nfritts is your team planning to work on this or should we close this issue to avoid users to have false expectations?
